[wasm] Bump the NoTargets version

lewing opened the pull request:
- Also, improve the CI triggers so `wasm-test-runner` does not trigger non-wasm jobs.

lewing committed:
Bump the NoTargets version

radical commented on src/tests/Common/wasm-test-runner/WasmTestRunner.proj:
```diff
@@ -1,6 +1,6 @@
 <!-- This project requires an explicit SDK version number because it is used on Helix,
       and global.json is not available. -->
-<Project Sdk="Microsoft.Build.NoTargets/1.0.53" DefaultTargets="WasmBuildApp">
+<Project Sdk="Microsoft.Build.NoTargets/3.5.6" DefaultTargets="WasmBuildApp">
```
We can drop the version here. The version is set in `global.json:14:    "Microsoft.Build.NoTargets": "3.5.0",`.
```suggestion
<Project Sdk="Microsoft.Build.NoTargets" DefaultTargets="WasmBuildApp">
```

lewing replied:
<!-- This project requires an explicit SDK version number because it is used on Helix,
      and global.json is not available. -->

radical committed:
Update src/tests/Common/wasm-test-runner/WasmTestRunner.proj

radical commented on src/tests/Common/wasm-test-runner/WasmTestRunner.proj:
```diff
@@ -1,6 +1,6 @@
 <!-- This project requires an explicit SDK version number because it is used on Helix,
       and global.json is not available. -->
-<Project Sdk="Microsoft.Build.NoTargets/1.0.53" DefaultTargets="WasmBuildApp">
+<Project Sdk="Microsoft.Build.NoTargets" DefaultTargets="WasmBuildApp">
```
Err we do need a version here because this project is built on helix, without a global.json . Setting this to match what we have in `global.json`.

```suggestion
<Project Sdk="Microsoft.Build.NoTargets/3.5.0" DefaultTargets="WasmBuildApp">
  <!-- Set the version for NoTargets here explicitly because this is built on helix, without a global.json -->
```

radical committed:
Update src/tests/Common/wasm-test-runner/WasmTestRunner.proj

radical committed:
evaluate-default-paths: add wasm-test-runner to wasm_specific_only subset, so it doesn't trigger non-wasm jobs

radical committed:
CI: update triggers

- wasm library tests, and build-only should be triggered by wasm changes
  only if there are non-runtime-tests changes, be runtimetest only
  changes already trigger the corresponding job

radical committed:
CI: don't trigger non-wasm jobs based on wasm pipeline changes

radical committed:
CI: don't trigger coreclr, or mono jobs based on only wasm-runtime tests changes

radical committed:
fix

radical committed:
fix yml .

lewing commented:
Failure is unrelated
